presto-oracle/src/main/java/com/facebook/presto/plugin/oracle/OracleClient.java

```diff
@@ -115,14 +115,19 @@ protected void renameTable(JdbcIdentity identity, String catalogName, SchemaTabl
             throw new PrestoException(NOT_SUPPORTED, "Table rename across schemas is not supported in Oracle");
         }
 
-        String newTableName = newTable.getTableName().toUpperCase(ENGLISH);
-        String oldTableName = oldTable.getTableName().toUpperCase(ENGLISH);
-        String sql = format(
-                "ALTER TABLE %s RENAME TO %s",
-                quoted(catalogName, oldTable.getSchemaName(), oldTableName),
-                quoted(newTableName));
-
         try (Connection connection = connectionFactory.openConnection(identity)) {
+            DatabaseMetaData metadata = connection.getMetaData();
+            String newTableName = newTable.getTableName();
+            String oldTableName = oldTable.getTableName();
+            if (metadata.storesUpperCaseIdentifiers() && !caseSensitiveNameMatchingEnabled) {
```

Comment:
Was the `storesUpperCaseIdentifiers` just missing for Oracle before or is there something different with it? Please add these Oracle changes explicitly to the PR description.

Reply:
OracleClient had renameTable implementation which had storesUpperCaseIdentifiers missing and converting table names to uppercase by default only for rename operation w/o check. So added it here. I can separate out this in different commit